compat/v1/compat.go

```diff
@@ -321,3 +323,90 @@ func QueryStateResponseFromState(state map[string]engine.XState) (QueryStateResp
 
 	return QueryStateResponse{State: stateResponse}, nil
 }
+
+type InT struct {
+	Struct []struct {
+		Txid    string `json:"txid,omitempty"`
+		Txindex string `json:"txindex,omitempty"`
+		Amount  string `json:"amount,omitempty"`
+		Payload string `json:"payload,omitempty"`
+		Phash   string `json:"phash,omitempty"`
+		To      string `json:"to,omitempty"`
+		Nonce   string `json:"nonce,omitempty"`
+		Nhash   string `json:"nhash,omitempty"`
+		Gpubkey string `json:"gpubkey,omitempty"`
+		Ghash   string `json:"ghash,omitempty"`
+	} `json:"struct"`
+}
+
+type InV struct {
+	Amount  string `json:"amount,omitempty"`
+	Ghash   string `json:"ghash,omitempty"`
+	Gpubkey string `json:"gpubkey,omitempty"`
+	Nhash   string `json:"nhash,omitempty"`
+	Nonce   string `json:"nonce,omitempty"`
+	Payload string `json:"payload,omitempty"`
+	Phash   string `json:"phash,omitempty"`
+	To      string `json:"to,omitempty"`
+	Txid    string `json:"txid,omitempty"`
+	Txindex string `json:"txindex,omitempty"`
+}
+
+type In struct {
+	T InT `json:"t"`
+	V InV `json:"v"`
+}
+
+type OutV struct {
+	Amount  string `json:"amount,omitempty"`
+	Hash    string `json:"hash,omitempty"`
+	Revert  string `json:"revert,omitempty"`
+	Sig     string `json:"sig,omitempty"`
+	Sighash string `json:"sighash,omitempty"`
+	Txid    string `json:"txid,omitempty"`
+}
+
+type OutT struct {
+	Struct []struct {
+		Hash    string `json:"hash,omitempty"`
+		Amount  string `json:"amount,omitempty"`
+		Sighash string `json:"sighash,omitempty"`
+		Sig     string `json:"sig,omitempty"`
+		Txid    string `json:"txid,omitempty"`
+		Revert  string `json:"revert,omitempty"`
+	} `json:"struct"`
+}
```

Comment:
I would've thought it would be simpler to just have an alternate `LockMintBurnReleaseOutput` type that does not contain a revert field, and we use that type if revert is empty. That way we wouldn't need to define all these "type" structs, and we could continue to use pack for encoding.